src/node/createSetupServer.ts

```diff
@@ -112,6 +112,7 @@ export function createSetupServer(...interceptors: Interceptor[]) {
           requestHandlers,
           ...nextHandlers,
         )
+        emitter.removeAllListeners()
```

Comment:
I wonder what's your thoughts on the expectations here. Imagine this setup:

```js
const server = setupServer()

beforeAll(() => {
  server.listen()
  server.on('request:start', callback)
})

afterEach(() => {
  // Calling this will now remove both "request:start" listeners.
  server.resetHandlers()
})

it('some test', () => {
  server.on('request:start', anotherCallback)
})
```

Would you expect `server.resetHandlers` to _wipe out_ all the listeners, including those you've added in the `beforeAll` hook?

The `.resetHandlers` method removes all runtime handlers (those you've added with `server.use()`) but preserves those you've initially passed to the `setupServer` call. I know it's a different thing, but I was wondering if people would expect it behave the same for event listeners in life-cycle events.

Reply:
That's a good point. Maybe need to add a separate method to allow the removal of the emitter's listeners.

Reply:
I've dwelled on this thought and came to the conclusion that coupling event listeners clean-up and `.resetHandlers()` isn't a good idea because it's not evident. I'm leaning towards having a separate method to remove event listeners from the `server`/`worker` instances. 

I also suggest to group the life-cycle events and the clean-up method under a single `events` property. The reason for that is `server.removeAllListeners()` being too ambiguous. 

```diff
-server.on('request:start', cb)
-worker.on('request:end', cb)
+server.events.on('request:start', cb)
+worker.events.on('request:end', cb)
```

Removing the listeners can then become something like:

```js
server.events.removeAllListeners()
```

I think we could also benefit from `server.events` being a regular `EventEmitter`. This would allow developers to remove certain listeners via `.removeListener(eventName, listener)` as they usually do with a regular event emitter.